pkg/pointerdb/pointerdb_test.go

```diff
@@ -85,6 +97,55 @@ func TestServicePut(t *testing.T) {
 	}
 }
 
+func makePointer(ctx context.Context, t *testing.T, numOfValidPieces, numOfInvalidPieces int) *pb.Pointer {
+	pieces := make([]*pb.RemotePiece, numOfValidPieces+numOfInvalidPieces)
+	for i := 0; i < numOfValidPieces; i++ {
+		identity, err := testidentity.NewTestIdentity(ctx)
+		assert.NoError(t, err)
+		pieces[i] = &pb.RemotePiece{
+			PieceNum: int32(i),
+			NodeId:   identity.ID,
+			Hash:     &pb.SignedHash{Hash: make([]byte, 32)},
+		}
+
+		_, err = rand.Read(pieces[i].Hash.Hash)
+		assert.NoError(t, err)
+		err = auth.SignMessage(pieces[i].Hash, *identity)
+		assert.NoError(t, err)
+	}
+
+	// public key did not match expected signer
+	for i := numOfValidPieces; i < len(pieces); i++ {
+		identity, err := testidentity.NewTestIdentity(ctx)
```

Comment:
`testplanet.NewPregeneratedIdentities()` to avoid creating so many identities in a test.